prog/encoding.go

```diff
@@ -317,6 +359,45 @@ func (p *parser) parseProg() (*Prog, error) {
 	return prog, nil
 }
 
+func (p *parser) parseCallProps() (CallProps, error) {
+	nameToValue := map[string]reflect.Value{}
+	callProps := DefaultCallProps()
+
+	callProps.ForeachProp(func(_, key string, value reflect.Value) {
+		nameToValue[key] = value
+	})
+	p.SkipWs()
+	for !p.EOF() && p.Char() != ')' {
+		propName := p.Ident()
+		value, ok := nameToValue[propName]
+		if !ok {
+			return callProps, fmt.Errorf("unknown call property: %s", propName)
```

Comment:
It's better to use `p.strictFailf` instead of returning an error. We try to parse program conservatively to preserve corpus, parse logs from mis-matching revisions, etc.

Reply:
Done, also added an extra test case to `TestDeserialize`.